plaidml/bridge/openvino/plaidml_builder.cpp

```diff
@@ -91,8 +93,12 @@ void ProgramBuilder::handleConstant(const std::shared_ptr<ngraph::Node>& node) {
 
 void ProgramBuilder::handleParameter(const std::shared_ptr<ngraph::Node>& node) {
   IE_ASSERT(node->get_output_size() == 1);
-  std::vector<int64_t> dims{node->get_shape().begin(), node->get_shape().end()};
-  plaidml::DType type = to_plaidml(node->get_element_type());
+  // TODO: Decide if we want to compare to the nGraph dims & type and issue warnings
+  // std::vector<int64_t> ng_dims{node->get_shape().begin(), node->get_shape().end()};
+  // plaidml::DType ng_type = to_plaidml(node->get_element_type());
+  auto inputDesc = networkInputs[node->get_friendly_name()]->getTensorDesc();
```

Comment:
Can't we get the tensor descriptor directly from the current node?

Reply:
The whole thing this PR is trying to fix is that IE can provide/request input/output tensors of different data types than the types of the Parameter and Result nodes in the nGraph network representing those tensors. I.e. a Result node might be fp16 but the IE TensorDesc for that node might be fp32; in this case we insert a cast to fp32 that is not present in the network graph.

Reply:
Yes, but I'm wondering if we can get the Tensor Descriptor from the node, and not look at the shape/type. But I could see how maybe the network info differs from the node and that they are out of sync.